### DISCUSSION TITLE
Night

### DISCUSSION THREAD

**Discussion by jamesdeluk** [chrisblakley/Daylight-Calendar-ICS#10] [Ideas]
Is it possible to create the opposite function, i.e. calendar entries showing night?

**Comment by chrisblakley** [comment]
It would definitely be possible, just a bit more tricky since it would need to account for the day number change at midnight. This is probably something that I'd be happy to point a developer in the right direction for, but I don't know if I'd have the capacity to maintain it myself.

If anyone wants to develop this either as part of the existing file, or a separate file and make a PR I'd be happy to share this repo for that.

I'll leave this open as an idea for anyone interested.

**Reply by jamesdeluk** [reply]
Sounds good! My knowledge of PHP is effectively zero so I won't be doing it either, but thanks for leaving this open.

**Reply by Elmeraculous** [reply]
Hi Chris. I ran into your great calendar. I have dusted off my github account especially so that I could comment on this RFC. This is exactly what I need for my use case to succeed. Which is send automatic out of office responses based on calendar events that are between sunset and sunrise (part of a game (Geocaching)). I am not a developer myself, so I probably won't be able to do this myself. I wouldn't mind donating a couple of bucks though if someone (or you Chris) could make this work. Thanks in advance!

**Reply by Elmeraculous** [reply]
I don't believe it. I scrolled through your license file because I was wondering whether or not I was allowed to have ChatGPT help me. I could not find anything in there saying that I couldn't so I decided to give it a go. ChatGPT spat out code in no time which I downloaded into a php file, uploaded it to my server and what do you know, it works... These times we live in.

Chris, how do you want to proceed from here. Is this 'nighttime.php' file something you would be interested in including in your branch? It's not polished yet (need to change the name of the Calendar from 'daylight' to 'nighttime' for example) but that's just dotting the i's. Please advise.

**Reply by chrisblakley** [reply]
Glad to hear you had success with that– you are definitely free to play around with the code and do whatever you want with it!

Feel free to make a PR after you've polished the code to your satisfaction and I'd be happy to include that file it in this repo alongside the daylight version.

**Reply by Elmeraculous** [reply]
Cool, will try to do so this week!

**Reply by Elmeraculous** [reply]
@chrisblakley Chris, my gitflow is a little rusty. I wanted to created a branch from your repo, but I can't seem to do so. I think I can create a fork though. Is that what you need me to do? And then create a pull request from that fork? I did create an issue just now (https://github.com/chrisblakley/Daylight-Calendar-ICS/issues/16) from the discussion topic, because I figured that would be the way to go in order to link it to my branch/fork. Please advise.

**Reply by chrisblakley** [reply]
That is correct– you can fork the entire repo, push your commit to that forked branch and then when it's ready you can make a pull request that I can merge here.

**Reply by Elmeraculous** [reply]
Done https://github.com/chrisblakley/Daylight-Calendar-ICS/pull/17
